### ISSUE TITLE
Programmatical Power button

### ISSUE THREAD

**Issue by peter-gasparovic** [dealfonso/power-buttons#13]
- add to Options -> onConfirm function that will be called when the confirmation button is clicked (instead of onclick).
- add to Options -> focusElement for focusing on an element in the confirmation window.

**Comment by dealfonso** [comment]
hi, are you suggesting to add those enhancements? 

I mean... in common usage, onConfirm will be the click event (i.e., it is confirmed). It will make sense to chain actions (e.g., call a function before the next power action is executed). But the click event will only be dispatched if the action is confirmed.

The option "focusElement" will make sense to me, so I will consider it for any action.

**Comment by peter-gasparovic** [comment]
hi, I use it like this

powerButtons('confirm', element, {
                confirm:  $(element).attr('data-title'),
                title: null,
                buttonConfirm: 'Yes',
                buttonCancel: 'No',
                escapeKey: true,
                // new feature that I need
                rootClass: 'powerbuttons-confirm' ->  would by generated in the template window, like this code "let dialog = appendToElement(createTag(".powerbuttons-confirm.modal.fade""
                onConfirm: function() { call after click confirm button }; -> now I must generate onclick atribute to make it work
                focusElement: '' -> CSS selector to focus element
            });


> hi, are you suggesting to add those enhancements?
> 
> I mean... in common usage, onConfirm will be the click event (i.e., it is confirmed). It will make sense to chain actions (e.g., call a function before the next power action is executed). But the click event will only be dispatched if the action is confirmed.
> 
> The option "focusElement" will make sense to me, so I will consider it for any action.

**Comment by dealfonso** [comment]
I do not fully understand the `rootClass` option...

**Comment by peter-gasparovic** [comment]
> I do not fully understand the `rootClass` option...

hi, 

only a more precise selector for styling the confirm dialog, if I depend only on the default class like .modal.show, something else could also affect such CSS, e.g. other dialog

for example CSS

/* --------------------------------------------------------- */
/* BEGIN - powerbuttons-confirm                              */

.powerbuttons-confirm.modal .modal-header {
    padding: .5rem !important;
}

.powerbuttons-confirm.modal .modal-body {
    font: 13px Arial, serif;
    padding: .5rem !important;
    white-space: pre-wrap;
}

.powerbuttons-confirm.modal .modal-footer {
    padding: .5rem !important;
}

.powerbuttons-confirm.modal .modal-footer .btn-secondary {
    padding: 3px 25px;
}

/* END - powerbuttons-confirm                                */
/* --------------------------------------------------------- */

**Comment by peter-gasparovic** [comment]
> I do not fully understand the `rootClass` option...

maybe a better name would be rootSelector

**Comment by dealfonso** [comment]
work in progress...

```javascript
DEFAULTS={
...
// The function to call when the action is confirmed (prior to the next action)
onConfirm: null,
// The class to apply to the dialog
dialogClass: "",
// The selector to focus when the dialog is shown
focus: "",
}
```

**Comment by peter-gasparovic** [comment]
> work in progress...
> 
> ```js
> DEFAULTS={
> ...
> // The function to call when the action is confirmed (prior to the next action)
> onConfirm: null,
> // The class to apply to the dialog
> dialogClass: "",
> // The selector to focus when the dialog is shown
> focus: "",
> }
> ```

it can be

**Comment by dealfonso** [comment]
Hi,
I have added the features that you suggested. 
1. the `dialogClass` and `focus` have been added to every action that shows a dialog. In the declarative way, you have the `data-*-dialog-class` and `data-*-focus` attributes
2. the `data-confirm-on-confirm` and `onConfirm` have been added to the _confirm_ action. So that it can be executed either if it is a string (they are executed using `eval`), or if it is a function.

You can check it at [the branch](https://github.com/dealfonso/power-buttons/tree/dialog-class). The doc has been updated

You can use the following URL to test the new options: `https://cdn.jsdelivr.net/gh/dealfonso/power-buttons@dialog-class/dist/powerbuttons.min.js`

**Comment by peter-gasparovic** [comment]
I'll try it, thanks.

**Comment by peter-gasparovic** [comment]
All the new features work perfectly, but you could also generate a *.map file for powerbuttons.min.js in your project, it's not acute, but it's possible.

Thank you very much for your time and I will respond so quickly.

**Comment by dealfonso** [comment]
Hi,

I added the creation of the map files for both the minified and the compressed dist files. Could you please check them?

**Comment by peter-gasparovic** [comment]
it's ok, thank you

**Comment by peter-gasparovic** [comment]
Hi,

I would still like to give you some tips for improvement, currently the window is centered in the middle, but something like [https://mdbootstrap.com/docs/standard/components/popconfirm], Inline positions (scroll down) would also be interesting.

**Comment by dealfonso** [comment]
thank you for the suggestions, and thank your for testing the changes

I have released version 2.2.0 which includes the enhancements

**Comment by peter-gasparovic** [comment]
ok, I already used it in a real project
